File: back/dentalium/src/main/java/com/romoreno/dentalium/infraestructure/adapter/in/controller/rest/appointment/dto/SearchAppointmentResponse.java
package com.romoreno.dentalium.infraestructure.adapter.in.controller.rest.appointment.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
public class SearchAppointmentResponse {

    private List<Appointment> appointmentList;

    public SearchAppointmentResponse() {
        appointmentList = new ArrayList<>();
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Appointment {
        private Integer id;
        private Date initDate;
        private Date endDate;
        private String observations;
        private Status status;
        private Person doctor;
        private Person patient;
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Person {
        private String id;
        private String name;
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Status {
        private String id;
        private String description;
    }

}

File: front/dentalium/src/app/model/api/SearchAppointmentResponse.ts
export class SearchAppointmentResponse {
  appointmentList: Appointment[];

  constructor() {
    this.appointmentList = [];
  }
}

export class Appointment {
  id: number;
  initDate: Date;
  endDate: Date;
  observations: string;
  status: Status;
  doctor: Person;
  patient: Person;

  constructor(
    id: number,
    initDate: Date,
    endDate: Date,
    observations: string,
    status: Status,
    doctor: Person,
    patient: Person
  ) {
    this.id = id;
    this.initDate = initDate;
    this.endDate = endDate;
    this.observations = observations;
    this.status = status;
    this.doctor = doctor;
    this.patient = patient;
  }
}

export class Person {
  id: string;
  name: string;

  constructor(id: string, name: string) {
    this.id = id;
    this.name = name;
  }
}

export class Status {
  id: string;
  description: string;

  constructor(id: string, description: string) {
    this.id = id;
    this.description = description;
  }
}
